### PR TITLE
crowdsec: init at 1.4.2

### LINKED ISSUE #155822: add crowdsec packages (agents and bouncer)
**Project description**
CrowdSec - the open-source and participative IPS able to analyze visitor behavior & provide an adapted response to all kinds of attacks. It also leverages the crowd power to generate a global CTI database to protect the user network.

**Metadata**

* homepage URL: https://crowdsec.net
* source URL: https://github.com/crowdsecurity/crowdsec 
* license: mit
* platforms: linux

### PR DESCRIPTION
**Pull request by urandom2**
###### Description of changes

Fixes #155822
Closes #162633

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by 06kellyjac** [comment]
What about bouncers?

**Comment by urandom2** [comment]
this change attempts to only package the binaries, there is no effort made to create nixos modules.

can you clarify what if anything is not working, preferably in a way that I can reproduce it locally? alternatively, can you clarify what you mean by bouncers?

File: pkgs/tools/security/crowdsec/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/crowdsecurity/crowdsec/pkg/cwversion.Version=v${version}"
+    "-X github.com/crowdsecurity/crowdsec/pkg/cwversion.BuildDate=1970-01-01_00:00:00"
+  ];
+
+  postBuild = "mv $GOPATH/bin/{crowdsec-cli,cscli}";
```
Comment (SuperSandro2000) [inline pkgs/tools/security/crowdsec/default.nix:30]: This should ideally be multi line

**Comment by 06kellyjac** [comment]
> this change attempts to only package the binaries, there is no effort made to create nixos modules. that can be completed as a later change.
> 
> can you clarify what if anything is not working, preferably in a way that I can reproduce it locally? alternatively, can you clarify what you mean by bouncers?

https://docs.crowdsec.net/docs/bouncers/intro

AFAIK the `crowdsec` binary is not useful without bouncers

`cscli` can interact with existing `crowdsec` deployments so I guess that can be immediately useful

**Pull request merged**